pkg/bgpv1/manager/reconciler/route_policy.go

```diff
@@ -361,3 +363,65 @@ func peerAddressFromPolicy(p *types.RoutePolicy) string {
 	}
 	return ""
 }
+
+// mergeAndDedupCommunities merges numeric standard community and well-known community strings,
+// deduplicated by their actual community values.
+func mergeAndDedupCommunities(standard []v2alpha1api.BGPStandardCommunity, wellKnown []v2alpha1api.BGPWellKnownCommunity) ([]string, error) {
+	var res []string
+	existing := make(map[uint32]struct{})
```

Comment:
I recently came across k8s set [package](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#Set). It seems like a nice alternative to map[]struct{}. See if it makes sense to use it here, as well as in `dedupLargeCommunities` method

Reply:
thanks for the suggestion, modified to use k8s `sets.Set`